server/etcdserver/apply.go

```diff
@@ -434,7 +434,15 @@ func (a *applierV3backend) Txn(ctx context.Context, rt *pb.TxnRequest) (*pb.TxnR
 		ctx = context.WithValue(ctx, traceutil.TraceKey, trace)
 	}
 	isWrite := !isTxnReadonly(rt)
-	txn := mvcc.NewReadOnlyTxnWrite(a.s.KV().Read(trace))
+
+	// When the transaction contains write operations, we use ReadTx instead of
+	// ConcurrentReadTx to avoid extra overhead of copying buffer.
+	var txn mvcc.TxnWrite
+	if isWrite {
```

Comment:
How about

```
// For read-only workloads, we use shared buffer by copying transaction read buffer
// for higher concurrency with ongoing blocking writes.
// For write/write-read transactions, we use the shared buffer
// rather than duplicating transaction read buffer.
```

Reply:
> How about
> 
> ```
> // For read-only workloads, we use shared buffer by copying transaction read buffer
> // for higher concurrency with ongoing blocking writes.
> // For write/write-read transactions, we use the shared buffer
> // rather than duplicating transaction read buffer.
> ```

sure